### ISSUE TITLE
Dockerised installation 

### ISSUE THREAD

**Issue by Adz0rd** [chris-belcher/electrum-personal-server#65]
Any chance that you could add a dockerfile for automating the installation of EPS + Bitcoin Core etc?

**Comment by Adz0rd** [comment]
Any idea for when this might materialise?

**Comment by chris-belcher** [comment]
I don't know.

**Comment by BobleChinois** [comment]
I gave it a shot, here's where I am up to:
* I created a docker image with Electrum Personal Server on Python3.7-alpine
* A docker-compose.yml file builds an image of Bitcoin Core and tweaks the parameter to make the two dockerized applications work together
The first step seems to work, but I'm having various issues when I try to make a docker compose, I'm a bit unexperienced with docker...
If someone can help me debug it I'll open a PR

**Comment by chris-belcher** [comment]
there appears to be a docker project here https://github.com/peleion/eps-docker

**Comment by BobleChinois** [comment]
Yes I find another similar here https://github.com/takinbo/eps-docker and here https://github.com/schulterklopfer/bitcoind_tor_eps_docker
What I was up to was more similar to the 2nd one, an easy way to set bitcoind and eps using docker. This look interesting I will have a closer look, but maybe it doesn't make sense to add this kind of stuff in this repo

**Comment by chris-belcher** [comment]
> maybe it doesn't make sense to add this kind of stuff in this repo

Why not? (I've never used docker before so please forgive if its a silly question)

**Comment by BobleChinois** [comment]
Don't worry I was just thinking out loud. It's not silly at all, and to answer it maybe we need to define what we want to achieve with a dockerized electrum personal server. In my view, it could be to : 
1. make the install of electrum personal server easier (not very useful, it's already pretty straightforward, and people that are technical enough to use docker are not very likely to have issues installing electrum personal server in the first place)
2. Make it easily portable between different environments (I'm not sure if this is really an issue since it's lightweight and don't have much dependencies)
3. Bundle it with other software like bitcoind, tor, btcpay etc to reduce the configuration overhead for users
The proposal made in btcpay repo was clearly 3, and I think that's the valid use case for electrum personal server.

What we can aim to achieve is to make an "official" docker image that people can directly pull for their own project (and keep it up to date), and eventually make some modifications that are needed like adding HD keys at runtime for btcpay. Here's what we should do then:
1. make a Dockerfile template on this repo
2. use this Dockerfile to build a "canonical" docker image and upload it to dockerhub
3. make sure that user can update `config.ini` at run time (seems to me to be the tricky part, I have to see how the others that made a dockerfile tackle this)
4. Add other ad hoc features, like the run time add HD keys for btcpay

Let me know what you think.

**Comment by maltokyo** [comment]
I would be really interested in EPS in a docker.  Any further thoughts on if this will happen here?
